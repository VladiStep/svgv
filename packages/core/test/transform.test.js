const { transform } = require('../src/index.js');
const { fixture } = require('./utils.js');

const paths = [undefined, null, '', 'foo/bar.svg'];
const defaultExports = [undefined, null, '', '"foo"', 'foo + "bar"'];

describe('transform', () => {
  it('should transform svg', async () => {
    const result = await transform(await fixture('raw.svg'));
    expect(result).toMatchSnapshot('result');

    for (const path of paths) {
      const actual = await transform(await fixture('raw.svg'), undefined, {
        path,
      });
      expect(actual).toEqual(result);
    }
  });

  it.each(defaultExports)(
    "should transform svg { defaultExport: '%s' }",
    async (defaultExport) => {
      const result = await transform(await fixture('raw.svg'), undefined, {
        defaultExport,
      });
      expect(result).toMatchSnapshot('result');

      for (const path of paths) {
        const actual = await transform(await fixture('raw.svg'), undefined, {
          path,
          defaultExport,
        });
        expect(actual).toEqual(result);
      }
    }
  );

  it('should throw error for invalid svg', async () => {
    await expect(transform(await fixture('invalid.svg'))).rejects.toThrow();
  });
});

describe('optimize', () => {
  it('should not optimize', async () => {
    const result = await transform(await fixture('raw.svg'), {
      svgo: false,
    });
    expect(result).toMatchSnapshot('result');
  });

  it('should allow options', async () => {
    const result = await transform(await fixture('raw.svg'), {
      svgo: {
        js2svg: {
          pretty: true,
          indent: 4,
        },
      },
    });
    expect(result).toMatchSnapshot('result');
  });

  it('should allow plugins', async () => {
    let result = await transform(await fixture('raw.svg'), {
      svgo: {
        plugins: ['removeViewBox'],
      },
    });
    expect(result).toMatchSnapshot('removeViewBox');

    result = await transform(await fixture('raw.svg'), {
      svgo: {
        plugins: null,
      },
    });
    expect(result).toMatchSnapshot('null');
  });
});
