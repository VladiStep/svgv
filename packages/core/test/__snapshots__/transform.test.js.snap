// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optimize should allow options: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\">
    <circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/>
</svg>
</template>
"
`;

exports[`optimize should allow plugins: null 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100px\\" height=\\"100px\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" style=\\"fill:#00f\\"/></svg></template>
"
`;

exports[`optimize should allow plugins: removeViewBox 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100px\\" height=\\"100px\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" style=\\"fill:#00f\\"/></svg></template>
"
`;

exports[`optimize should not optimize: result 1`] = `
"<template><?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\" \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100px\\" height=\\"100px\\">
  <circle class=\\"circle\\" cx=\\"50\\" cy=\\"50\\" r=\\"50\\" />
  <style type=\\"text/css\\">
  .circle {
    fill: blue;
  }
  .unknown {
    fill: red;
  }
  </style>
  <script>console.error('svg');</script>
</svg>
</template>
"
`;

exports[`transform should transform svg { defaultExport: '"foo"' }: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/></svg></template>
<script>export default new String(\\"foo\\");</script>
"
`;

exports[`transform should transform svg { defaultExport: '' }: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/></svg></template>
"
`;

exports[`transform should transform svg { defaultExport: 'foo + "bar"' }: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/></svg></template>
<script>export default new String(foo + \\"bar\\");</script>
"
`;

exports[`transform should transform svg { defaultExport: 'null' }: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/></svg></template>
"
`;

exports[`transform should transform svg { defaultExport: 'undefined' }: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/></svg></template>
"
`;

exports[`transform should transform svg: result 1`] = `
"<template><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"50\\" fill=\\"#00f\\"/></svg></template>
"
`;
